Stop installing monitoring, we are going to delete it.

**Describe the bug**

I noticed in a CI run that Eventing seems to be installing `https://storage.googleapis.com/knative-nightly/serving/latest/monitoring.yaml`.  We deprecated this in 0.14, and plan to remove this, so eventing's prow jobs will be broken.

 **Expected behavior**

Don't install deprecated things.

 **To Reproduce**

N/A

 **Knative release version**

HEAD

 **Additional context**

I noticed this due to a `gke-resource-quotas` flake here: https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_eventing/4273/pull-knative-eventing-conformance-tests/1315286177904857089

Bonus: fewer moving parts means fewer flakes 😉 

cc @grantr @vaikas @lionelvillard

`TestBrokerTracing` and `TestChannelTracingWithReply` seem to have a dependency on this, they fail with:
```
util.go:108: Error while setup Zipkin tracing: error while parsing the Zipkin endpoint in config-tracing config map: could not extract namespace/name from
```

When monitoring isn't installed.

--- a/.github/workflows/kind-e2e.yaml
+++ b/.github/workflows/kind-e2e.yaml
@@ -24,8 +24,7 @@ jobs:
 
         test-suite:
         - ./test/e2e
-        # TODO(https://github.com/knative/eventing/issues/4275): Enable this.
-        # - ./test/conformance
+        - ./test/conformance
 
         # Map between K8s and KinD versions.
         # This is attempting to make it a bit clearer what's being tested.
@@ -44,10 +43,8 @@ jobs:
         # Add the flags we use for each of these test suites.
         - test-suite: ./test/e2e
           extra-test-flags: -brokerclass=MTChannelBasedBroker -channels=messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel,messaging.knative.dev/v1:Channel,messaging.knative.dev/v1:InMemoryChannel -sources=sources.knative.dev/v1alpha2:ApiServerSource,sources.knative.dev/v1alpha2:ContainerSource,sources.knative.dev/v1alpha2:PingSource
-        # TODO(https://github.com/knative/eventing/issues/4275): Enable this.
-        # - test-suite: ./test/conformance
-        #   extra-test-flags: -brokers=eventing.knative.dev/v1beta1:MTChannelBasedBroker -channels=messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel,messaging.knative.dev/v1:Channel,messaging.knative.dev/v1:InMemoryChannel -sources=sources.knative.dev/v1beta1:ApiServerSource,sources.knative.dev/v1alpha2:ContainerSource,sources.knative.dev/v1beta1:PingSource
-
+        - test-suite: ./test/conformance
+          extra-test-flags: -brokers=eventing.knative.dev/v1beta1:MTChannelBasedBroker -channels=messaging.knative.dev/v1beta1:Channel,messaging.knative.dev/v1beta1:InMemoryChannel,messaging.knative.dev/v1:Channel,messaging.knative.dev/v1:InMemoryChannel -sources=sources.knative.dev/v1beta1:ApiServerSource,sources.knative.dev/v1alpha2:ContainerSource,sources.knative.dev/v1beta1:PingSource
     env:
       GOPATH: ${{ github.workspace }}
       GO111MODULE: off

--- a/test/conformance/helpers/broker_data_plane_test_helper.go
+++ b/test/conformance/helpers/broker_data_plane_test_helper.go
@@ -34,7 +34,7 @@ import (
 )
 
 var podMeta = metav1.TypeMeta{
-	Kind:       "pod",
+	Kind:       "Pod",
 	APIVersion: "v1",
 }
 

--- a/test/conformance/helpers/tracing_test_helper.go
+++ b/test/conformance/helpers/tracing_test_helper.go
@@ -65,14 +65,17 @@ func tracingTest(
 	tracinghelper.Setup(t, client)
 	clusterDomain := tracinghelper.GetClusterDomain(t, client)
 
-	// Setup the test infrastructure
-	expectedTestSpan, eventMatcher := setupInfrastructure(ctx, t, &channel, client, recordEventsPodName, true, clusterDomain)
-
-	// Start the event info store and assert the event was received correctly
+	// Start the event info store. Note this is done _before_ we setup the infrastructure, which
+	// sends the event.
 	targetTracker, err := recordevents.NewEventInfoStore(client, recordEventsPodName, client.Namespace)
 	if err != nil {
 		t.Fatal("Pod tracker failed:", err)
 	}
+
+	// Setup the test infrastructure
+	expectedTestSpan, eventMatcher := setupInfrastructure(ctx, t, &channel, client, recordEventsPodName, true, clusterDomain)
+
+	// Assert that the event was seen.
 	matches := targetTracker.AssertAtLeast(1, recordevents.MatchEvent(eventMatcher))
 
 	// Match the trace

--- a/test/lib/k8s_events.go
+++ b/test/lib/k8s_events.go
@@ -35,6 +35,8 @@ type EventListener struct {
 
 	lock     sync.Mutex
 	handlers []EventHandler
+
+	eventsSeen int
 }
 
 // NewEventListener creates a new event listener
@@ -59,7 +61,9 @@ func NewEventListener(client kubernetes.Interface, namespace string, logf func(s
 
 	go func() {
 		eventsInformer.Run(ctx.Done())
-		logf("EventListener stopped")
+		el.lock.Lock()
+		defer el.lock.Unlock()
+		logf("EventListener stopped, %v events seen", el.eventsSeen)
 	}()
 
 	return &el
@@ -68,15 +72,19 @@ func NewEventListener(client kubernetes.Interface, namespace string, logf func(s
 func (el *EventListener) handle(event *corev1.Event) {
 	el.lock.Lock()
 	defer el.lock.Unlock()
+	el.eventsSeen++
 	for _, handler := range el.handlers {
 		handler(event)
 	}
 }
 
-func (el *EventListener) AddHandler(handler EventHandler) {
+func (el *EventListener) AddHandler(handler EventHandler) int {
 	el.lock.Lock()
 	defer el.lock.Unlock()
 	el.handlers = append(el.handlers, handler)
+	// Return the number of events that have already been seen. This helps debug scenarios where
+	// the expected event was seen, but only before this handler was added.
+	return el.eventsSeen
 }
 
 func (el *EventListener) Stop() {

--- a/test/lib/operation.go
+++ b/test/lib/operation.go
@@ -72,7 +72,7 @@ func (c *Client) WaitForResourceReadyOrFail(name string, typemeta *metav1.TypeMe
 		if untyped != nil {
 			c.T.Errorf("Object that did not become ready %v-%s when dumping error state: %+v", *typemeta, name, untyped)
 		}
-		c.T.Fatalf("Failed to get %v-%s ready: %+v", *typemeta, name, errors.WithStack(err))
+		c.T.Fatalf("Failed to get %s-%s ready: %+v", typemeta, name, errors.WithStack(err))
 	}
 }
 

--- a/test/lib/recordevents/event_info.go
+++ b/test/lib/recordevents/event_info.go
@@ -58,12 +58,16 @@ func (ei *EventInfo) String() string {
 type SearchedInfo struct {
 	TotalEvent int
 	LastNEvent []EventInfo
+
+	storeEventsSeen    int
+	storeEventsNotMine int
 }
 
 // Pretty print the SearchedInfor for error messages
 func (s *SearchedInfo) String() string {
 	var sb strings.Builder
-	sb.WriteString(fmt.Sprintf("%d events seen, last %d events:", s.TotalEvent, len(s.LastNEvent)))
+	sb.WriteString(fmt.Sprintf("%d events seen, last %d events (total events seen %d, events ignored %d):",
+		s.TotalEvent, len(s.LastNEvent), s.storeEventsSeen, s.storeEventsNotMine))
 	for _, ei := range s.LastNEvent {
 		sb.WriteString(ei.String())
 		sb.WriteRune('\n')

--- a/test/lib/recordevents/event_info_store.go
+++ b/test/lib/recordevents/event_info_store.go
@@ -61,6 +61,9 @@ type EventInfoStore struct {
 
 	lock      sync.Mutex
 	collected []EventInfo
+
+	eventsSeen    int
+	eventsNotMine int
 }
 
 // Creates an EventInfoStore that is used to iteratively download events recorded by the
@@ -72,7 +75,8 @@ func NewEventInfoStore(client *testlib.Client, podName string, podNamespace stri
 		podNamespace: podNamespace,
 	}
 
-	client.EventListener.AddHandler(store.handle)
+	numEventsAlreadyPresent := client.EventListener.AddHandler(store.handle)
+	client.T.Logf("EventInfoStore added to the EventListener, which has already seen %v events", numEventsAlreadyPresent)
 
 	return store, nil
 }
@@ -84,8 +88,12 @@ func (ei *EventInfoStore) getEventInfo() []EventInfo {
 }
 
 func (ei *EventInfoStore) handle(event *corev1.Event) {
+	ei.lock.Lock()
+	defer ei.lock.Unlock()
+	ei.eventsSeen += 1
 	// Filter events
 	if !ei.isMyEvent(event) {
+		ei.eventsNotMine += 1
 		return
 	}
 
@@ -96,8 +104,6 @@ func (ei *EventInfoStore) handle(event *corev1.Event) {
 		return
 	}
 
-	ei.lock.Lock()
-	defer ei.lock.Unlock()
 	ei.collected = append(ei.collected, eventInfo)
 }
 
@@ -120,7 +126,12 @@ func (ei *EventInfoStore) Find(matchers ...EventInfoMatcher) ([]EventInfo, Searc
 	f := AllOf(matchers...)
 	const maxLastEvents = 5
 	allMatch := []EventInfo{}
-	sInfo := SearchedInfo{}
+	ei.lock.Lock()
+	sInfo := SearchedInfo{
+		storeEventsSeen:    ei.eventsSeen,
+		storeEventsNotMine: ei.eventsNotMine,
+	}
+	ei.lock.Unlock()
 	lastEvents := []EventInfo{}
 	var nonMatchingErrors []error
 

